lib/registrations.js

```diff
@@ -182,27 +166,29 @@ function buildNewClassFunction (Type) {
  * @return {Function}
  * The function used for dependency resolution
  */
-function generateResolve (fn, depsFn, isFunction) {
+function generateResolve (fn, dependencyParseTarget) {
+  // If the function used for dependency parsing is null, use the supplied function
+  if (!dependencyParseTarget) {
+    dependencyParseTarget = fn
+  }
+
   // Use a regular function instead of an arrow function to facilitate binding to the container object
   return function (container) {
     // Because the container holds a global reolutionMode we need to determine it in the proper order or precedence:
     // registration -> container -> default value
-    let resolutionMode = (this.resolutionMode || container.__resolutionMode || ResolutionModes.PROXY)
+    let resolutionMode = (this.resolutionMode || container.options.resolutionMode || ResolutionModes.PROXY)
```

Comment:
Not sure if this is enforced in the linter, but if you don't modify a variable after initialization, you should use `const`, it communicates intent and the reader will know that it won't be modified. 😄  I can fix these no problem, just letting you know.